carmm/analyse/meshgrid/meshgrid_functions.py

```diff
@@ -0,0 +1,153 @@
+    """
+    Function finds the distance between a point (defined in cartesian co-ordinates a_xx, a_yy, a_zz)
+    relative to all points on a Numpy meshgrid (defined in the unit_cell_object).
+    Args:
+        a_xx: float
+            x coordinate of input point.
+        a_yy: float
+            y coordinate of input point.
+        a_zz: float
+            z coordinate of input point.
+        MeshObject: MeshgridObject
+            Object storing meshgrid and PBC conditions
+    Returns:
+        mesh_distances: Numpy meshgrid of distances from point [a_xx, a_yy, a_zz].
+    """
+
+    import numpy as np
+
+    frac_a = np.dot(np.array([a_xx, a_yy, a_zz]),MeshObject.inverse_cell_array)
+
+    x_vec_dist = np.abs(frac_a[0] - MeshObject.frac_xx)
+    y_vec_dist = np.abs(frac_a[1] - MeshObject.frac_yy)
+    z_vec_dist = np.abs(frac_a[2] - MeshObject.frac_zz)
+
+    if MeshObject.pbc[0]==1:
+        x_vec_dist = np.where(x_vec_dist > 0.5, x_vec_dist - 1., x_vec_dist)
+    if MeshObject.pbc[1]==1:
+        y_vec_dist = np.where(y_vec_dist > 0.5, y_vec_dist - 1., y_vec_dist)
+    if MeshObject.pbc[2]==1:
+        z_vec_dist = np.where(z_vec_dist > 0.5, z_vec_dist - 1., z_vec_dist)
+
+    # Convert mesh distances back to cartesian coordinates.
+    stack_mesh = np.stack((x_vec_dist, y_vec_dist, z_vec_dist), axis=-1)
+    stack_mesh = np.einsum('ji,abcj->jabc', MeshObject.Cell.array, stack_mesh)
+
+    mesh_distances = np.linalg.norm(stack_mesh, axis=0)
+
+    return mesh_distances
+
+def distance_point2point(x_1, y_1, z_1, x_2, y_2, z_2, MeshObject):
+    """
+    Function finds the distance between two points (defined in cartesian co-ordinates).
+    Args:
+        x_1: float
+            x coordinate of point 1.
+        y_1: float
+            y coordinate of point 1.
+        z_1: float
+            z coordinate of point 1.
+        x_2: float
+            x coordinate of point 2.
+        y_2: float
+            y coordinate of point 2.
+        z_2: float2
+            z coordinate of point 2.
+        MeshObject: MeshgridObject
+            Object storing meshgrid and PBC conditions
+    Returns:
+        o_distance: Distance between points 1 and 2.
+    """
+
+    from ase.geometry import get_distances
+
+    o_distance = get_distances([x_1, y_1, z_1], p2=[x_2, y_2, z_2],
+                                cell=MeshObject.Cell, pbc=MeshObject.pbc)
+
+    return o_distance
+
+def midpoint_points(x_1, y_1, z_1, x_2, y_2, z_2, MeshObject):
+    """
+    Function finds the distance between two points (defined in cartesian co-ordinates).
+    Args:
+        x_1: float
+            x coordinate of point 1.
+        y_1: float
+            y coordinate of point 1.
+        z_1: float
+            z coordinate of point 1.
+        x_2: float
+            x coordinate of point 2.
+        y_2: float
+            y coordinate of point 2.
+        z_2: float2
+            z coordinate of point 2.
+        MeshObject: MeshgridObject
```

Comment:
Is the change in naming scheme necessary?

Reply:
Not at all - resolved.